src/documentation/programming-guide.md

```diff
@@ -1079,13 +1052,185 @@ For blocking execution, append the `waitUntilFinish` method:
+
+> NOTE: If you create your `PCollection` from in-memory data by using the `Create` transform, you cannot rely on coder inference and default coders. `Create` does not have access to any typing information for its arguments, and may not be able to infer a coder if the argument list contains a value whose exact run-time class doesn't have a default coder registered. 
+
+{:.language-java}
+When using `Create`, the simplest way to ensure that you have the correct coder is by invoking `withCoder` when you apply the `Create` transform.
+
+#### Default coders and the CoderRegistry
+
+Each Pipeline object has a `CoderRegistry` object, which maps language types to the default coder the pipeline should use for those types. You can use the `CoderRegistry` yourself to look up the default coder for a given type, or to register a new default coder for a given type.
+
+{:.language-java}
+`CoderRegistry` contains a default mapping of coders to standard Java types for any Pipeline you create using the Beam SDK for Java. The following table shows the standard mapping:
+
+{:.language-java}
+| Java Type        | Default Coder         |
+|------------------|-----------------------|
+| Double           | DoubleCoder           |
+| Instant          | InstantCoder          |
+| Integer          | VarIntCoder           |
+| Iterable         | IterableCoder         |
+| KV               | KvCoder               |
+| List             | ListCoder             |
+| Map              | MapCoder              |
+| Long             | VarLongCoder          |
+| String           | StringUtf8Coder       |
+| TableRow         | TableRowJsonCoder     |
+| Void             | VoidCoder             |
+| byte[ ]          | ByteArrayCoder        |
+| TimestampedValue | TimestampedValueCoder |
+
+
+{:.language-py}
+`CoderRegistry` contains a default mapping of Coders to standard Python types for any Pipeline you create using the Beam SDK for Python. The following table shows the standard mapping:
+
+{:.language-py}
+| Python Type | Default Coder |
+|-------------|---------------|
+| int         | VarIntCoder   |
+| float       | FloatCoder    |
+| str         | BytesCoder    |
+| bytes       | StrUtf8Coder  |
+| Tuple       | TupleCoder    |
+
+
+
+##### Looking up a default coder
+
+{:.language-java}
+You can use the method CoderRegistry.getDefaultCoder to determine the default Coder for a Java type. You can access the CoderRegistry for a given Pipeline by using the method Pipeline.getCoderRegistry. This allows you to determine (or set) the default Coder for a Java type on a per-pipeline basis: i.e. "for this pipeline, verify that Integer values are encoded using BigEndianIntegerCoder."
```

Comment:
verify that -> use?

(And is there a reason Py doesn't have the same example?)